.github/workflows/openxla_pin_update_weekly.yml

```diff
@@ -0,0 +1,79 @@
+name: Weekly Code Update PR
+on:
+  workflow_dispatch: # Manual trigger
+  schedule:
+    - cron: '0 9 * * 1'  # Each Monday 9:00AM (PDT time)
+
+jobs:
+  update_and_create_pr:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Get current date
+        run: echo "CURRENT_DATE=$(date +'%Y%m%d')" >> $GITHUB_ENV
+
+      - name: Create new branch
+        run: git checkout -b weekly-update-${{ env.CURRENT_DATE }}
+
+      - name: Install sed
+        run: sudo apt-get update && sudo apt-get install -y sed
+
+      - name: Get JAX version for specified date
+        run: |
+          JAX_VERSION=$(curl -s https://storage.googleapis.com/jax-releases/jax_releases.json | jq -r '.[] | select(.released_on == "${{ env.CURRENT_DATE }}") | .version')
+          if [ -z "$JAX_VERSION" ]; then
+            echo "No JAX release found for ${{ env.CURRENT_DATE }}. Skipping update."
+            exit 0
+          fi
+          echo "JAX_VERSION=$JAX_VERSION" >> $GITHUB_ENV
+
+      - name: Update date in setup.py
+        run: |
+          sed -i "s/_date =.*/_date = '${{ env.CURRENT_DATE }}'/" ${FILE_PATH}
+          sed -i "s/_jax_version =.*/_jax_version = f'${{ env.JAX_VERSION }}.dev${{ env.CURRENT_DATE }}'/" ${FILE_PATH}
+          git add ${FILE_PATH}
+          # git commit -m "Weekly update setup.py to (${{ env.CURRENT_DATE }})"
+          # git push origin HEAD
+        env:
+          FILE_PATH: setup.py
+
+      - name: Fetch HEAD commit from OpenXLA github repo
+        run: |
+          git remote add target https://github.com/openxla/xla.git
+          git fetch target
+          echo "HEAD_COMMIT=$(git rev-parse target/main)" >> $GITHUB_ENV
+
+      - name: Update OpenXLA commit in WORKSPACE
+        run: |
+          sed -i "s/xla_hash =.*/xla_hash = '${{ env.GITHUB_ENV }}'/" ${FILE_PATH}
+          git add ${FILE_PATH}
+          git commit -m "Weekly update WORKSPACE with (${{ env.GITHUB_ENV }})"
+          git push origin HEAD
+        env:
+          FILE_PATH: WORKSPACE
+
+      - name: Create Pull Request
+        uses: actions/create-pull-request@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          branch: weekly-update
+          base: main
+          title: "Weekly Code Update - ${{ env.CURRENT_DATE }}"
+          body: "This PR contains the weekly OpenXLA-pin update at commit: (HEAD commit: ${{ env.HEAD_COMMIT }}), with libtpu version at ${{ env.CURRENT_DATE }}."
+
+      - name: Add TPUCI label
```

Comment:
`create-pull-request` has a `labels` option: https://github.com/marketplace/actions/create-pull-request#action-inputs

If the label is not on the PR when it is created, the TPU CI will not run.

Reply:
good suggestion, have we used `create-pull-request` before? wanna confirm before use

Reply:
thanks, added labels `tpuci` in current PR, please correct me if I understand wrong

Reply:
If the `tpuci` label is not on the PR when it is created, nothing will happen. I haven't use `create-pull-request` myself before, but I'm just going off of the documentation.  Adding `labels: tpuci` to the step above should work.

Reply:
Thanks, updated above with `labels: tpuci`

Reply:
You can remove this last step since the PR will be created with that label